src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -253,21 +253,10 @@ private PodTemplateCallback(PodTemplate podTemplate) {
          * Remove the template after step is done
          */
         protected void finished(StepContext context) throws Exception {
-            Cloud cloud = Jenkins.get().getCloud(cloudName);
-            if (cloud == null) {
-                LOGGER.log(Level.FINE, "Cloud {0} no longer exists, cannot delete pod template {1}",
-                        new Object[] { cloudName, podTemplate.getName() });
-                return;
-            }
-            if (cloud instanceof KubernetesCloud) {
-                LOGGER.log(Level.FINE, "Removing pod template {1} from cloud {0}",
-                        new Object[] { cloud.name, podTemplate.getName() });
-                KubernetesCloud kubernetesCloud = (KubernetesCloud) cloud;
-                kubernetesCloud.removeDynamicTemplate(podTemplate);
-            } else {
-                LOGGER.log(Level.WARNING, "Cloud is not a KubernetesCloud: {0} {1}",
-                        new String[] { cloud.name, cloud.getClass().getName() });
-            }
+            KubernetesCloud cloud = resolveCloud();
```

Comment:
Will this fail the build if the cloud is not found? Is that the desired affect? I guess it would alert the user of an issue.

Reply:
This would happen only if the cloud is removed while running the step. I guess it is highly unlikely to happen (unless you're actively changing the instance configuration, and in that case it wouldn't be unexpected to see some jobs failing as a result of removing the kubernetes cloud).